### DISCUSSION TITLE
Contributing in 2023...

### DISCUSSION THREAD

**Discussion by dreaming-augustin** [artyom-beilis/cppcms#98] [General]
Hello Artyom,

It is very nice to see you here again.

I am a long time cppcms user, but I remain an amateur programmer as I do this part time only.
I lack the knowledge and experience to tackle big difficult code development.

Having said that, if cppcms is still maintained and active, I would be happy to help out with simple tasks, tidying code, testing, etc, if pull requests are integrated into the main branches.

I will be busy during the next couple of months, but if the project is active, I'd be happy to make some time in 2023 to contribute something back.

Maybe you can label specific issues as junior tasks...

**Comment by dreaming-augustin** [comment]
I should be available starting now for the next few months. If anybody replies to this thread I hope I'll see it in time.
I renew my offer to help, provided I receive a minimum of guidance and the result gets committed either in the main repository, or in a maintained fork.

**Comment by artyom-beilis** [comment]
Glad to hear.

I'm actually using CppCMS quite actively lately in my other FOSS project: https://github.com/artyom-beilis/OpenLiveStacker
What areas do you want to contribute to?

My general thoughts are:

- WebSockets
- SSL support

**Comment by dreaming-augustin** [comment]
Hi! :smile:  

Do not overestimate my skills. As I wrote in my introduction (above), I am only a part-time amateur developer. I do not have the skills to come up with brand new features.

I can do small tasks like:
- code review and code testing of pull requests, if it helps getting the pull requests getting committed into the main repository.
- minor fixes (like English typos, etc.)
- code style (trailing blank spaces, etc.)
- test development branches for functionality.
- any other junior tasks that you think I could handle.

I use Linux (Gentoo) with GCC.

**Comment by dreaming-augustin** [comment]
I don't know if it's relevant, but I'm going to start using WebRTC for my application.

**Comment by dreaming-augustin** [comment]
cppdb needs maintenance as well. The master branch no longer works with MySQL 8. 

What is the official repository that you use and accept pull requests for?

cppdb on github #71 

https://github.com/melpon/cppdb
https://github.com/alexbodn/cppdb

cppdb and MySQL 8 #70

**Comment by artyom-beilis** [comment]
> cppdb needs maintenance as well.

That would be cool.

I'll migrate cppdb to github instead of SF.

**Comment by artyom-beilis** [comment]
https://github.com/artyom-beilis/cppdb Cloned it :-)

Build and realized that there is so much work to do. It needs C++11, replace `auto_ptr` with `unique_ptr`, replace home-built mutex and atomic counter with std::mutex and std::atomic.

Add move ctor/assignment to most classes.

Maybe even check possibility of replacement of `ref_ptr` with `shared_ptr` (but this one is way more tricky and less nessary)
